code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -280,8 +281,16 @@ LINEN BINS
 
 /obj/item/bedsheet/random/Initialize()
 	..()
-	var/type = pick(typesof(/obj/item/bedsheet) - /obj/item/bedsheet/random)
-	new type(loc)
+	var/list/sheets = list(typesof(/obj/item/bedsheet) - /obj/item/bedsheet/random)
+	sheets = shuffle(sheets)
+	var/obj/item/bedsheet/sheet
+	for(var/i in sheets)
+		sheet = new i(loc)
+		if(sheet.randomizable)
```

Comment:
typecast `i` and use `initial(i.randomizable)` and create the bedsheet on the `break` instead of creating and qdeling unneeded sheets

Reply:
huh, didn't realize you could check values of type paths